kusto/ingest/ingest.go

```diff
@@ -266,6 +267,17 @@ func ReportResultToTable() FileOption {
 	)
 }
 
+// SetCreationTime option allows the user to override the data creation time the retention policies are considered against
+// If not set the data creation time is considered to be the time of ingestion
+func SetCreationTime(creationTime time.Time) FileOption {
```

Comment:
Changed the argument to t.
About validation - Not a fan of forcing current and future server side validations.

Reply:
Up to you on the validation, as Kusto has very little in terms of validations on the server side.  If you mispelled the property it will work, but not like you expect.  No idea what it does when you are out of time range.  Fixing things on the client side is certainly not the greatest, but if you can't control the server......

The other thing to note is that you have no test here to make sure this sets anything in the message.  At minimum I'd ask that you test your implementation against the server and validate the entries went in with the correct time.  With a normal service, the service would tell us that we sent something broken.  As Kusto tends not to do this, it would be good if we could make sure this works for the release.

Reply:
Tested this locally one and it works :)